File: Python/1047 - Tempo de Jogo com Minutos.py
# author: Josélio Júnior (Lunatic Fox) <joseliojrx25@gmail.com>
# copyright: Josélio Júnior (Lunatic Fox) 2022

import math
ih, im, eh, em = map(lambda x: int(x), input().split(' '))
p = lambda h, m: print(f'O JOGO DUROU {str(h)} HORA(S) E {str(m)} MINUTO(S)' )

tm = (ih * 60 + im) - (eh * 60 + em)
if ih == eh and im == em:
  p(24, 0)
elif ih == eh:
  mm = im - em
  p(23, 60 - mm) if mm > 0 else p(0, mm * -1)
else:
  m = 60 - (tm % 60)
  if tm > 0:
    p(math.floor((1440 - tm) / 60), 0 if m == 60 else m)
  else:
    p(math.ceil(tm / 60) * -1, 0 if m == 60 else m)

File: Pascal/1047 - Tempo de Jogo com Minutos.pas
// author: Josélio Júnior (Lunatic Fox) <joseliojrx25@gmail.com>
// copyright: Josélio Júnior (Lunatic Fox) 2023

uses sysutils, math;
const
  H = 1440;
var
  rd, ptt, sn: string;
  tms: array[1..4] of integer;
  i, tm, ih, im, eh, em, m: integer;
procedure p(h, m: integer);
begin
  writeln('O JOGO DUROU ', inttostr(h), ' HORA(S) E ', inttostr(m), ' MINUTO(S)');
end;
begin
  readln(rd);
  for i := 1 to 4 do
  begin
    if i < 4 then
    begin
      ptt := copy(rd, 1, pos(' ', rd));
      sn := stringreplace(ptt, ' ', '', []);
      tms[i] := strtoint(sn);
      rd := stringreplace(rd, ptt, '', []);
    end
    else
      tms[i] := strtoint(rd);
  end;

  ih := tms[1];
  im := tms[2];
  eh := tms[3];
  em := tms[4];

  tm := (ih * 60 + im) - (eh * 60 + em);
  if (ih = eh) and (im = em) then
    p(24, 0)
  else
  begin
    if ih = eh then
    begin
      m := im - em;
      if m > 0 then
        p(23, 60 - m)
      else
        p(0, m * -1);
    end
    else
    begin
      if tm > 0 then
      begin
        m := 60 - (tm mod 60);
        if m = 60 then
          p(floor((H - tm) / 60), 0)
        else
          p(floor((H - tm) / 60), m);
      end
      else
      begin
        m := abs(tm mod 60);
        if m = 60 then
          p(ceil(tm / 60) * -1, 0)
        else
          p(ceil(tm / 60) * -1, m);
      end;
    end;
  end;
end.
